agent/eni/statemanager/statemanager_linux.go

```diff
@@ -16,159 +16,85 @@
-	if _, ok := enis[mac]; !ok {
-		log.Warnf("ENI state manager: device with MACAddress: %s missing from managed state", mac)
-		return
+		return eni, true
 	}
 
-	delete(enis, mac)
+	log.Error("ENI state manager: device with empty mac address")
+	return nil, false
 }
 
-// RemoveDeviceUnsafe is used to remove new devices from uDev events
-// NOTE: removeDeviceUnsafe expects lock to be held prior to update for correct semantics
-func (statemanager *stateManager) RemoveDeviceUnsafe(deviceName string) {
-	log.Debugf("ENI state manager: removing device: %s (unsafe)", deviceName)
-
-	enis := statemanager.GetAll()
-	for mac, dev := range enis {
-		if dev == deviceName {
-			statemanager.RemoveDeviceWithMACAddressUnsafe(mac)
-			return
-		}
+// HandleENIEvent handles the eni event from udev or reconcil phase
+func (statemanager *stateManager) HandleENIEvent(mac string) {
+	eni, ok := statemanager.ENIStateChangeShouldBeSent(mac)
+	if ok {
+		eni.Status = api.ENIAttached
+		statemanager.emitENIAttachmentEvent(api.TaskStateChange{
+			TaskArn:     eni.TaskArn,
+			Attachments: eni,
+		})
 	}
-	log.Debug("ENI state manager: no device was removed from the map")
 }
 
-// AddDeviceWithMACAddress adds new devices upon initialization
-func (statemanager *stateManager) AddDeviceWithMACAddress(deviceName, macAddress string) {
-	log.Debugf("ENI state manager: adding device %s with MAC %s", deviceName, macAddress)
-
-	statemanager.updateLock.Lock()
-	defer statemanager.updateLock.Unlock()
-
-	statemanager.AddDeviceWithMACAddressUnsafe(deviceName, macAddress)
+// emitENIAttachmentEvent send the eni statechange(attach) to event handler
+func (statemanager *stateManager) emitENIAttachmentEvent(event api.TaskStateChange) {
+	log.Infof("ENI state manager: sending eni state change to event handler: %v", event)
+	statemanager.taskEvent <- event
 }
 
-// RemoveDeviceWithMACAddress is used to remove new devices from maintained state
-func (statemanager *stateManager) RemoveDeviceWithMACAddress(mac string) {
-	log.Debugf("ENI state manager: removing device with MACAddress: %s", mac)
-
-	statemanager.updateLock.Lock()
-	defer statemanager.updateLock.Unlock()
-
-	statemanager.RemoveDeviceWithMACAddressUnsafe(mac)
-}
-
-// RemoveDevice is used to remove new devices from uDev events
-func (statemanager *stateManager) RemoveDevice(deviceName string) {
-	log.Debugf("ENI state manager: removing device: %s", deviceName)
-
-	statemanager.updateLock.Lock()
-	defer statemanager.updateLock.Unlock()
-
-	statemanager.RemoveDeviceUnsafe(deviceName)
-}
-
-func (statemanager *stateManager) IsMACAddressPresent(macAddress string) bool {
-	log.Debugf("ENI state manager: checking state for MACAddress: %s", macAddress)
-
-	statemanager.updateLock.RLock()
-	defer statemanager.updateLock.RUnlock()
-
-	_, ok := statemanager.enis[macAddress]
-	return ok
+// Reconcile performs a 2 phase reconciliation of managed state
+func (statemanager *stateManager) Reconcile(currentState map[string]string) {
```

Comment:
Why did you get rid of everything but these lines in this method? Can you please modify the documentation for this method and the comments in the code to reflect the same?